esp/services/ws_dfu/ws_dfuService.cpp

```diff
@@ -1901,10 +1901,24 @@ void CWsDfuEx::doGetFileDetails(IEspContext &context, IUserDescriptor* udesc, co
     }
     //#14280
 
-    FileDetails.setWuid(df->queryAttributes().queryProp("@workunit"));
+    const char* wuid = df->queryAttributes().queryProp("@workunit");
+    FileDetails.setWuid(wuid);
     if (version >= 1.28)
         FileDetails.setNumParts(df->numParts());
 
+    if (cluster && *cluster && wuid && *wuid && (version >= 1.31))
+    {
+        Owned<IDFUWorkUnitFactory> factory = getDFUWorkUnitFactory();
+        Owned<IDFUWorkUnit> wu = factory->updateWorkUnit(wuid);
```

Comment:
This isn't right..

You are looking to see *if* the file was created by a dfu spray job and then looking at the workunit and it's replication..

As I explained in the parent issue (HPCC-13853), you need to get the ClusterPartDiskMapSpec from the file itself (IFileDescriptor).